### PR TITLE
chore: add relevant events logging

### PR DESCRIPTION
**Pull request by JoCa96**
A new Storybook Util `withNativeEventLoggingFor` is introduced.
It generates Storybook `ArgTypes` that allow us to document and log relevant native events in Storybook.
These can be merged with other `ArgTypes`.
The native events will be documented in a new section "Relevant HTML Events".

Relates to #1603

### CODE REVIEW

File: .changeset/tender-zebras-flow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@sit-onyx/storybook-utils": minor
+---
+
+feat: added withNativeEventLogging to log and document native events
```
Comment (larsrickert) [inline .changeset/tender-zebras-flow.md:5]: Should we also document this utility in our docs?

File: packages/storybook-utils/src/actions.ts [hunk 2]
```diff
@@ -23,6 +23,40 @@ export const enhanceEventArgTypes: ArgTypesEnhancer = ({ argTypes }) => {
...
+ * ```
+ *
+ * @param relevantEvents a list of event names that should be logged
+ * @returns Storybook ArgTypes object
+ */
+export const withNativeEventLogging = (relevantEvents: (keyof Events)[]) =>
+  relevantEvents.reduce((argTypes, eventName) => {
+    const action = eventName.replace(/^on/, "").toLowerCase();
+    argTypes[eventName] = {
+      name: eventName,
```
Comment (larsrickert) [inline packages/storybook-utils/src/actions.ts:49]:
```suggestion
      name: action,
```

Would prefer to show the event with the on prefix

File: packages/storybook-utils/src/actions.ts [hunk 3]
```diff
@@ -23,6 +23,40 @@ export const enhanceEventArgTypes: ArgTypesEnhancer = ({ argTypes }) => {
...
+      control: false,
+      description: `[${action}_event](https://developer.mozilla.org/en-US/search?q=element+${action}+event)`, // unfortunately there is no static path for all events
```
Comment (larsrickert) [inline packages/storybook-utils/src/actions.ts:51]:
```suggestion
      description: `Native HTML [${action}](https://developer.mozilla.org/en-US/search?q=element+${action}+event) event`, // unfortunately there is no static path for all events
```

Comment (larsrickert) [inline packages/storybook-utils/src/actions.ts:51]: (optional) we could maintain a map of "real" links for very common events like click, change, focus etc. and use the generic search link for all other events, what do you think?

If its too much manual maintenance we don't have to do it :)

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/storybook-utils/src/actions.ts (hunk 3 above)
Comment (JoCa96) [reply packages/storybook-utils/src/actions.ts]: Done, I had some fun with typescript :D

File: packages/storybook-utils/src/events.ts [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+ * Removes a prefix from a string type.
+ * @example ```ts
+ * type Trimmed = TrimStart<"onclick", "on">;
+ * const trimmed: Trimmed = "click";
+ * ```
+ */
+type TrimStart<
+  TString extends string,
+  Prefix extends string,
+> = TString extends `${Prefix}${infer Base}` ? Base : TString;
```
Comment (larsrickert) [inline packages/storybook-utils/src/events.ts:28]:
```suggestion
  TPrefix extends string,
> = TString extends `${TPrefix}${infer TBase}` ? TBase : TString;
```

Isn't it a convention to prefix generic types with T?

File: packages/storybook-utils/src/events.ts [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+
+/**
+ * The doc type ensures that a name has a valid URL that includes the name.
+ */
+type Docs<Name extends string> = {
+  name: Name;
+  url: `${string}${Name}${string}`;
+};
+
+export const EVENT_DOC_MAP: EventDocMap = {
```
Comment (larsrickert) [inline packages/storybook-utils/src/events.ts:45]: Nice TypeScript magic :D
Couldn't we map the event.name from the key? If I see correctly, the EVENT_DOC_MAP is always the event.name with the `on` prefix

**Review by larsrickert — CHANGES_REQUESTED** [review]
👏

File: packages/storybook-utils/src/events.ts (hunk 5 above)
Comment (larsrickert) [reply packages/storybook-utils/src/events.ts:45]: I meant the keys from our own `EVENT_DOC_MAP` :) So when we use it in the Storybook util, we could extract the event name from the key so we don't need to maintain them manually

**Review by larsrickert — APPROVED** [review]

**Pull request merged**